File: lsp_diagnostics_demo/broken.java
public class broken {
    public static void main(String[] args) {
        int x = "hello";  // type mismatch
        System.out.println(unknownVar);  // undefined symbol
        String name = 42  // missing semicolon

        if (true {  // missing closing paren
            System.out.println("test");
        }
    }
}

File: lsp_diagnostics_demo/broken.php
<?php

function add($a, $b) {
    return $a + $b
}  // missing semicolon on return

$name = "world"
echo "Hello, " . $undefinedVar;  // undefined variable

class MyClass {
    public function getValue( {  // missing closing paren
        return $this->value;
    }
}
